test/functional/specs/Personalization/C17409728.js

```diff
@@ -0,0 +1,371 @@
+
+  const propositionId = uuid();
+  const itemId = uuid();
+  const responseBody = createMockResponse({
+    propositionId,
+    items: [
+      {
+        id: itemId,
+        schema: "https://ns.adobe.com/personalization/dom-action",
+        data: {
+          type: "insertAfter",
+          content: '<div id="something-else">This is something else</div>',
+          selector: "#page-header",
+        },
+      },
+    ],
+  });
+
+  await addHtmlToBody(testPageBody, true);
+
+  const alloy = createAlloyProxy();
+  await alloy.configure(config);
+  await alloy.applyResponse({
+    renderDecisions: true,
+    responseBody,
+  });
+
+  await t
+    .expect(
+      Selector("#something-else").withAttribute(INTERACT_ID_DATA_ATTRIBUTE)
+        .exists,
+    )
+    .notOk();
+
+  edgeEndpointLogs.clear();
+
+  await t.click("#something-else");
+  await t.wait(REASONABLE_WAIT_TIME);
+
+  await t.expect(edgeEndpointLogs.count(() => true)).eql(0);
+});
+
+test("Test C17409728: Automatically sends interact event for proposition click when autoCollectPropositionInteractions configured to 'decoratedElementsOnly'", async () => {
+  config = compose(orgMainConfigMain, debugEnabled, {
+    autoCollectPropositionInteractions: {
+      [ADOBE_JOURNEY_OPTIMIZER]: DECORATED_ELEMENTS_ONLY,
+    },
+  });
+
+  const propositionId = uuid();
+  const itemId = uuid();
+  const responseBody = createMockResponse({
+    propositionId,
+    items: [
+      {
+        id: itemId,
+        schema: "https://ns.adobe.com/personalization/dom-action",
+        data: {
+          type: "insertAfter",
+          content: `<ul id="some-list">
+                      <li id="some-list-item-a">one</li>
+                      <li id="some-list-item-b">two</li>
+                      <li id="some-list-item-c">three <span id="some-list-item-span" ${CLICK_LABEL_DATA_ATTRIBUTE}="click-me-here">yippie</span></li>
+                    </ul>`,
+          selector: "#page-header",
+        },
+      },
+    ],
+  });
+
+  await addHtmlToBody(testPageBody, true);
+
+  const alloy = createAlloyProxy();
+  await alloy.configure(config);
+  await alloy.applyResponse({
+    renderDecisions: true,
+    responseBody,
+  });
+
+  await t
+    .expect(
+      Selector("#some-list").withAttribute(INTERACT_ID_DATA_ATTRIBUTE).exists,
+    )
+    .ok();
+
+  edgeEndpointLogs.clear();
+
+  await t.click("#some-list-item-a");
+
+  await responseStatus(edgeEndpointLogs.requests, [200, 204]);
+  await t.expect(edgeEndpointLogs.count(() => true)).eql(0);
+
+  await t.click("#some-list-item-span");
+
+  await responseStatus(edgeEndpointLogs.requests, [200, 204]);
+  await t.expect(edgeEndpointLogs.count(() => true)).eql(1);
+
+  // TODO: Testcafe no longer captures the request body for sendBeacon requests.
+  // const notification = JSON.parse(edgeEndpointLogs.requests[0].request.body);
+  //
+  // await t.expect(notification.events[0].xdm.eventType).eql(INTERACT);
+  //
+  // await t
+  //   .expect(
```

Comment:
Please let me know if there is a way to force `fetch` instead of `sendBeacon` for `collect` calls.  I tried to find a way but didn't see a clear path.